src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/DataProviderPass.php

```diff
@@ -55,6 +56,11 @@ private function registerDataProviders(ContainerBuilder $container, string $type
             foreach ($tags as $attributes) {
                 $priority = $attributes['priority'] ?? 0;
                 $queue->insert(new Reference($serviceId), $priority);
+
+                $definition = $container->getDefinition($serviceId);
```

Comment:
we should do this in the above loop no? I think this loops on the tags because a data provider can be both collection and item. There you're changing the definition twice, but once is enough.